File: random/isAnagram.ts

function isAnagram(strA: string, strB: string): boolean {
    const rearrangeString = (str: string): string => {
        /**
         * convert to a array
         * sort the array
         * convert the array to a list and return the re-arranged string
         */
        return [...str].sort((a, b) => a.localeCompare(b)).join('')
    }

    // an anagram string must have the same length
    if (strA.length !== strB.length) {
        return false;
    }

    return rearrangeString(strA) === rearrangeString(strB);
}

console.log(isAnagram("anagram", "nagaram"))

File: python/strings/is_anagram.py

def isAnagram(s: str, t: str) -> bool:
    def rearrangeString(_s: str):
        # re-arrange in ascending order
        newString = sorted(_s)
        # convert to string and return the re-arranged string
        return ''.join(newString)

    # an anagram string must have the same length
    return False if len(s) != len(t) else rearrangeString(s) == rearrangeString(t)


print(isAnagram(s="anagram", t="nagaram"))
